lib/bridge/MatrixHandler.js

```diff
@@ -1261,7 +1263,52 @@ MatrixHandler.prototype._sendIrcAction = Promise.coroutine(
         return;
     }
 
+    /**
+     * Cache events in here so we can refer to them for replies.
+     * We deliberately avoid fetching events from the homeserver due to speed concerns.
+     */
+    this._eventCache.set(event.event_id, { body: event.content.body, sender: event.sender });
+    if (this._eventCache.size > EVENT_CACHE_SIZE) {
+        const delKey = this._eventCache.entries().next().value[0];
+        this._eventCache.delete(delKey);
+    }
+
     let text = event.content.body;
```

Comment:
there's no read from the cache?

Reply:
also if we cache miss, we should do a hit to `/event`. Synapse's cache is likely going to be faster and more useful than ours anyways.

Reply:
>Synapse's cache is likely going to be faster and more useful than ours anyways.

I doubt it's going to be faster.

Reply:
https://github.com/matrix-org/matrix-appservice-irc/blob/6e0ac53ca578b1bf353e97a2d75f624b8612377a/lib/bridge/MatrixHandler.js#L1286

Reply:
Synapse's cache itself is faster. The HTTP overhead obviously won't be good.

Still should have a cache miss lookup. Put it behind an option if you'd like.

The event body is not looked up from the cache, therefore making the unsafe reliance on the sender's event still there.

Reply:
The JS SDK doesn't support the /event endpoint, which is going to really bloat this PR out. I've made the caching configurable anyway.

Reply:
It'd be great to have a js-sdk PR for it then. I'm quite surprised the endpoint doesn't already exist, given it's age and usefulness. Having a cache miss is important to deal with to prevent unexpected and unpredictable messages being sent, particularly by those who would be unaware of the problem in the first place.